tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,157 @@
+			Type:   "Reconciled",
+			Status: "True",
+		}
+
+		resourceReady = resourcecondition.Condition{
+			Type:   "Ready",
+			Status: "True",
+		}
+	)
+
+	Describe("operator installation", func() {
+		// TODO: we  need to support testing both types of deployment for the operator, helm and olm via subscription.
+		// Discuss with the team if we should add a flag to the test to enable the olm deployment and don't do that deployment in different step
+		When("default helm manifest are applied", func() {
+			BeforeAll(func() {
+				if skipDeploy == "true" {
+					Skip("Skipping the deployment of the operator and the tests")
+				}
+				GinkgoWriter.Println("Deploying Operator using default helm charts located in /chart folder")
+				Eventually(deployOperator).Should(Succeed(), "Operator deployment should be successful")
+			})
+
+			Specify("the operator is running", func() {
+				Eventually(kubectl.GetResourceCondition).WithArguments(namespace, "deployment", deploymentName).Should(ContainElement(resourceAvailable))
+				GinkgoWriter.Println("Operator deployment is Available")
+
+				Eventually(kubectl.GetPodPhase).WithArguments(namespace, "control-plane=istio-operator").Should(Equal("Running"), "Istio-operator pod should be Running")
+				GinkgoWriter.Println("Istio-operator pod is Running")
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio resource", func() {
+		for _, version := range istioVersions {
+			// Note: This var version is needed to avoid the closure of the loop
+			version := version
+
+			Context(fmt.Sprintf("is applied the istio resource with version %s", version), func() {
+				BeforeAll(func() {
+					Expect(kubectl.CreateNamespace(controlPlaneNamespace)).To(Succeed())
+					Eventually(createIstioCR).WithArguments(version).Should(Succeed(), "Istio CR should be created")
+					GinkgoWriter.Println("Istio CR created")
+				})
+
+				AfterAll(func() {
+					GinkgoWriter.Println("Cleaning up")
+					Eventually(kubectl.DeleteNamespace).WithArguments(controlPlaneNamespace).Should(Succeed(), "Namespace should be deleted")
+					Eventually(kubectl.CheckNamespaceExist).WithArguments(controlPlaneNamespace).Should(MatchError(kubectl.ErrNotFound), "Namespace should be deleted")
+					GinkgoWriter.Println("Cleanup done")
+				})
+
+				When("the Istio resource is created", func() {
+					It("updates the Istio resource status to Ready and Running", func() {
+						Eventually(kubectl.GetResourceCondition).WithArguments(controlPlaneNamespace, "istio", istioName).Should(ContainElement(resourceReconcilied))
+						Eventually(kubectl.GetResourceCondition).WithArguments(controlPlaneNamespace, "istio", istioName).Should(ContainElement(resourceReady))
+						Eventually(kubectl.GetPodPhase).WithArguments(controlPlaneNamespace, "app=istiod").Should(Equal("Running"), "Istiod should be Running")
+					})
+
+					Specify("the istio resource version match the applied version", func() {
```

Comment:
We should try to use `It` whenever possible, since it makes the test easier to read.

Here, we could rephrase this to:
```suggestion
					It("deploys correct istiod image tag according to the version in the Istio CR", func() {
```

Reply:
Also, this should be moved after `It("deploys istiod")`.